wake_t/beamline_elements.py

```diff
@@ -907,6 +916,9 @@ def _gamma(self, px, py, pz):
         return np.sqrt(1 + px**2 + py**2 + pz**2)
 
     def calculate_density(self, z):
+        if self.dens_func is not None:
+            return self.dens_func(z)
+
```

Comment:
Could you move this after the next `if`?  That would make sure that the behavior is consistent when the ramp is an `upramp`.

Reply:
I thought that in case of a function one does not need to specify whether if is downramp or upramp. 
Do I miss something?

Reply:
It's because right now the `z` coordinate is defined as the distance to the top of the ramp. In the case of a downramp, this is the beginning of the ramp, but in case of an upramp it is actually the end. This allows you to use the same function both for an up- and downramp without having to do any changes.

This might be redesigned in the future, but for the moment I would move these lines just to keep the consistency. Imagine you give as input profile the same exponential ramp that is already defined in wake-t. One would expect the same behavior as setting `profile='exponential'`, but right now this wouldn't be the case if an `upramp` is used.

Reply:
Ok, I understand that even for the case of a callable function one could still use the feature `upramp` to invert the direction automatically... Good! I put these lines after the next `if`. Thanks!

Reply:
Perfect, thanks!